pkg/assets/store.go

```diff
@@ -292,6 +294,34 @@ func (s *Store) AddSigV4(ctx context.Context, ns string, sigv4 *monitoringv1.Sig
 	return nil
 }
 
+// AddAzureOAuth processes the AzureOAuth SecretKeySelectors and adds the AzureOAuth data to the store.
+func (s *Store) AddAzureOAuth(ctx context.Context, ns string, azureAD *monitoringv1.AzureAD, key string) error {
+	if azureAD == nil {
+		return nil
+	}
+
+	azureOAuth := azureAD.OAuth
+	if azureOAuth == nil || azureOAuth.ClientSecret == nil {
+		return nil
+	}
+
+	if azureOAuth.ClientSecret == nil {
```

Comment:
This part of the code answer my question [here](https://github.com/prometheus-operator/prometheus-operator/pull/6037/files#r1369228320) you can indeed remove the `*`and then you don't need to check if it's nil